pkg/registry/decorated_registry.go

```diff
@@ -0,0 +1,111 @@
+package registry
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"golang.org/x/time/rate"
+)
+
+type DecoratedRegistryConfig struct {
+	RateLimiter DecoratedRegistryConfigRateLimiter
+	Underlying  Interface
+}
+
+type DecoratedRegistryConfigRateLimiter struct {
+	ListRepositories *rate.Limiter
+	ListTags         *rate.Limiter
+	Pull             *rate.Limiter
+	Push             *rate.Limiter
+}
+
+type DecoratedRegistry struct {
+	rateLimiter DecoratedRegistryConfigRateLimiter
+	underlying  Interface
+}
```

Comment:
It isn't used yet. I'm preparing the code for running in infinite loops. The idea is to wrap `Registry` implementing `Interface` without any code changes in the former.